### PR TITLE
rererefactor frontend files

### PR DESCRIPTION
**Pull request by pngwn**
## Description

Refactors the frontend files (yet again) to make things simpler.

Component packages have a maximum of two entry points:

- `Index.svelte` - this is the component (not optional)
- `Example.svelte` - this is the example component (optional)

By convention any other source files are store in a `/shared` folder. I have also removed all the re-exporting `index.ts` files because things were just getting confusing.

This has simplified many packages and many components.

- The layout components like `Group`, `Form` are most just a single file now.
- The form input components now have a much simplified interactive/static switching (just a prop)

This should make things much less overwhelming for custom component authors using our packages as templates.

`Image` is still an absolute disaster obviously.

Closes: #(issue)

## 🎯 PRs Should Target Issues

Before your create a PR, please check to see if there is [an existing issue](https://github.com/gradio-app/gradio/issues) for this change. If not, please create an issue before you create this PR, unless the fix is very small. 

Not adhering to this guideline will result in the PR being closed. 

## Tests

1. PRs will only be merged if tests pass on CI. To run the tests locally, please set up [your Gradio environment locally](https://github.com/gradio-app/gradio/blob/main/CONTRIBUTING.md) and run the tests: `bash scripts/run_all_tests.sh`

2. You may need to run the linters: `bash scripts/format_backend.sh` and `bash scripts/format_frontend.sh`

### CODE REVIEW

File: js/fileexplorer/Index.svelte [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+			{...loading_status}
+			autoscroll={gradio.autoscroll}
+			i18n={gradio.i18n}
+		/>
+
+		<DirectoryExplorer
+			bind:value
+			{file_count}
+			{server}
+			mode="interactive"
```
Comment (freddyaboulton) [inline js/fileexplorer/Index.svelte:97]: Do we need the if/else clause here. We can have one branch with {mode}, no?

Comment (pngwn) [reply js/fileexplorer/Index.svelte:97]: Yes! I missed this one.

File: js/uploadbutton/Index.svelte [hunk 2]
```diff
@@ -62,20 +63,39 @@
...
+	<UploadButton
+		{elem_id}
+		{elem_classes}
+		{visible}
+		{file_count}
+		{file_types}
+		{size}
+		{scale}
+		{min_width}
+		disabled
```
Comment (freddyaboulton) [inline js/uploadbutton/Index.svelte:76]: I think we can condense the code with disabled={mode === "static"}?

**Review by freddyaboulton — COMMENTED** [review]
This is great @pngwn !

The deployed demo is failing. Maybe it can't import state component?

image

Also, I noticed some of the unit test files still import from `static`, e.g. Audio. Would be good to fix for the person who will get CI green for the UI.

Other than that, should be good to merge!

**Comment by pngwn** [comment]
>The deployed demo is failing. Maybe it can't import state component?

Yeah thats just a casing issue, should be fixed in the latest commit (when it deploys).

File: js/uploadbutton/Index.svelte (hunk 2 above)
Comment (pngwn) [reply js/uploadbutton/Index.svelte:76]: Will check!

Comment (pngwn) [reply js/uploadbutton/Index.svelte]: fixd

**Comment by pngwn** [comment]
Should be ok now @freddyaboulton

**Review by freddyaboulton — APPROVED** [review]

**Pull request merged**